src/ifcwrap-csharp/IfcOpenShell.Net/IfcOpenShell.Net/EntityInstanceExtensions.cs

```diff
@@ -0,0 +1,391 @@
+        }
+        
+        public static bool TryGetAttributeAsEntity(this EntityInstance instance , string attributeName, out EntityInstance attributeValue)
+        {
+            attributeValue = default;
+            return instance.GetAttribute(attributeName)?.TryGetAsEntity(out attributeValue) ?? false;
+        }
+        
+        public static bool TryGetAttributeAsIntList(this EntityInstance instance , string attributeName, out IntVector attributeValue)
+        {
+            attributeValue = default;
+            return instance.GetAttribute(attributeName)?.TryGetAsIntList(out attributeValue) ?? false;
+        }
+        
+        public static bool TryGetAttributeAsDoubleList(this EntityInstance instance , string attributeName, out DoubleVector attributeValue)
+        {
+            attributeValue = default;
+            return instance.GetAttribute(attributeName)?.TryGetAsDoubleList(out attributeValue) ?? false;
+        }
+        
+        public static bool TryGetAttributeAsStringList(this EntityInstance instance , string attributeName, out StringVector attributeValue)
+        {
+            attributeValue = default;
+            return instance.GetAttribute(attributeName)?.TryGetAsStringList(out attributeValue) ?? false;
+        }
+        
+        public static bool TryGetAttributeAsEntityList(this EntityInstance instance , string attributeName, out aggregate_of_instance attributeValue)
+        {
+            attributeValue = default;
+            var att = instance.GetAttribute(attributeName);
+            return att?.TryGetAsEntityList(out attributeValue) ?? false;
+        }
+        #endregion
+        
+        public static EntityArgument? TryGetAttributeAtIndex(this EntityInstance? instance, uint key)
+        {
+            if (instance == null) return null;
+            if(key >= instance.Length())
+            {
+                throw new IndexOutOfRangeException($"Attribute index {key} out of range for instance of type {instance.Is()}");
+            }
+            return instance.get_argument(key).TryGetAsEntity();
+        }
+
+        public static ArgumentResult GetArgument(this EntityInstance instance, uint argIndex)
+        {
+            return new ArgumentResult.FromArgumentByType(instance.get_argument(argIndex));
+        }
+        
+        public static ArgumentResult? GetAttribute(this EntityInstance instance, string attributeName) {
+            // attribute categories:
+            const int INVALID = 0;
+            const int FORWARD = 1;
+            const int INVERSE = 2;
+            
+            // 0 == INVALID, 1 == FORWARD, 2 == INVERSE
+            var attributeCategory = instance.get_attribute_category(attributeName);
+            if (attributeCategory == FORWARD)
+            {
+                var idx = instance.get_argument_index(attributeName);
+                // TODO need to implement weird pointer lookup thing
```

Comment:
Maybe be more specific *what* weird pointer lookup thing is needed, or where you can refererence for instance python code that contains "weird pointer lookup thing"